python/pyspark/worker.py

```diff
@@ -879,6 +940,8 @@ def verify_result(result):
                 verify_pandas_result(
                     result, return_type, assign_cols_by_name=False, truncate_return_schema=False
                 )
+                for result_tuple in result.itertuples():
+                    check_output_row_against_schema(list(result_tuple))
```

Comment:
Shall we move this to before the pandas DataFrame is created?

Reply:
I tried that originally but the UDTF results is an Iterable and it turns out that iterating through it consumes the values, making it impossible to create the DataFrame after because the iterator is empty :)